internal/icinga2/api_responses.go

```diff
@@ -135,6 +157,31 @@ type HostServiceRuntimeAttributes struct {
 	AcknowledgementLastChange UnixFloat   `json:"acknowledgement_last_change"`
 }
 
+// MarshalLogObject implements the zapcore.ObjectMarshaler interface.
+func (hsra *HostServiceRuntimeAttributes) MarshalLogObject(encoder zapcore.ObjectEncoder) error {
+	encoder.AddString("name", hsra.Name)
+	if hsra.Host != "" {
+		encoder.AddString("host", hsra.Host)
+	}
+
+	encoder.AddInt("state", hsra.State)
+	encoder.AddInt("state_type", hsra.StateType)
+	encoder.AddBool("in_downtime", hsra.DowntimeDepth != 0)
+	encoder.AddBool("acknowledged", hsra.Acknowledgement != AcknowledgementNone)
+	encoder.AddTime("last_state_change", hsra.LastStateChange.Time())
+	err := encoder.AddArray("groups", zapcore.ArrayMarshalerFunc(func(encoder zapcore.ArrayEncoder) error {
+		for _, group := range hsra.Groups {
+			encoder.AppendString(group)
+		}
+		return nil
+	}))
+	if err != nil {
+		return err
+	}
+
+	return encoder.AddObject("check_result", &hsra.LastCheckResult)
```

Comment:
Again, could you try to keep the same order as for the struct fields? And why was only `AcknowledgementLastChange` excluded?

Reply:
> Again, could you try to keep the same order as for the struct fields?

Same thing here too! The check result is a separate json object, and I was just trying to reorder things so they can be read easily, just like you wouldn't want to add the `zap.Error()` field before something else, but always as the last one.

> And why was only AcknowledgementLastChange excluded?

Same here as well! I don't see any use case where this could be helpful for troubleshooting.

Reply:
For me its also the same again :D